src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt

```diff
@@ -107,31 +100,10 @@ class PersonService(
 
   private fun isCreateEvent(event: String?) = listOf(PRISONER_CREATED, NEW_OFFENDER_CREATED).contains(event)
 
-  fun findCandidateRecords(person: Person): Page<PersonEntity> {
-    val postcodeSpecifications = person.addresses.map { PersonSpecification.levenshteinPostcode(it.postcode) }
-
-    val soundexFirstLastName = Specification.where(
-      PersonSpecification.soundex(person.firstName, PersonSpecification.FIRST_NAME)
-        .and(PersonSpecification.soundex(person.lastName, PersonSpecification.LAST_NAME)),
-    )
-
-    val levenshteinDob = Specification.where(PersonSpecification.levenshteinDate(person.dateOfBirth, PersonSpecification.DOB))
-    val levenshteinPostcode = Specification.where(PersonSpecification.combineSpecificationsWithOr(postcodeSpecifications))
-
-    return readWriteLockService.withReadLock {
-      personRepository.findAll(
-        Specification.where(
-          exactMatch(person.sourceSystemType.name, SOURCE_SYSTEM).and(
-            exactMatch(person.otherIdentifiers?.pncIdentifier?.toString(), PNC)
-              .or(exactMatch(person.driverLicenseNumber, DRIVER_LICENSE_NUMBER))
-              .or(exactMatch(person.nationalInsuranceNumber, NI))
-              .or(exactMatch(person.otherIdentifiers?.croIdentifier?.toString(), CRO))
-              .or(soundexFirstLastName.and(levenshteinDob.or(levenshteinPostcode))),
-          ),
-        ),
-        Pageable.ofSize(PAGE_SIZE),
-      )
-    }
+  fun searchForRecord(person: Person): PersonEntity? {
+    val pageablePersonEntities: Page<PersonEntity> = searchService.findCandidateRecords(person)
+    val personEntity: PersonEntity? = searchService.processCandidateRecords(pageablePersonEntities.content, person)
+    return personEntity
```

Comment:
This is nicer now. 
The first call to `searchService` gets back `pageablePersonEntities` which is only used to pass into another call to `searchService` - mabe this could just be one method which returns a personEntity?

Reply:
Yeah i thought this, but this makes the separate components testing i.e. findCandidateRecord easier. Unless you know a way of testing private functions.